virttest/nfs.py

```diff
@@ -332,6 +332,66 @@ def setup_remote(self):
+            if not status:
+                firewall_services = output.split(':')[1].strip().split(' ')
+                if 'nfs' not in firewall_services:
+                    service_cmd = "firewall-cmd --permanent --zone=public "
+                    service_cmd += "--add-service=nfs"
+                    s_status, s_output = commands.getstatusoutput(service_cmd)
+                    if s_status:
+                        logging.error("nfs service not added in firewall: "
+                                      "%s" % s_output)
+                    else:
+                        firewalld.restart()
+                        logging.debug("nfs service added to firewall "
+                                      "sucessfully")
+                else:
+                    logging.debug("nfs service already permitted by firewall")
+            else:
+                # For RHEL 6 based system firewall-cmd is not available
+                logging.debug("Using iptables to permit NFS service")
+                nfs_ports = []
+                nfsd = service.Factory.create_service("nfs")
+                rpcd = service.Factory.create_service("rpcbind")
+                iptables = service.Factory.create_service("iptables")
+                nfs_sysconfig = params.get("nfs_sysconfig_path",
+                                           "/etc/sysconfig/nfs")
+                tcp_port = params.get("nfs_tcp_port", "32803")
+                udp_port = params.get("nfs_udp_port", "32769")
+                mountd_port = params.get("nfs_mountd_port", "892")
+                subnet_mask = params.get("priv_subnet", "192.168.2.0/24")
+                nfs_ports.append("LOCKD_TCPPORT=%s" % tcp_port)
+                nfs_ports.append("LOCKD_UDPPORT=%s" % udp_port)
+                nfs_ports.append("MOUNTD_PORT=%s" % mountd_port)
+                cmd_output = process.system_output("cat %s" % nfs_sysconfig,
+                                                   shell=True)
+                exist_ports = cmd_output.strip().split('\n')
+                # check if the ports are already configured, if not then add it
+                for each_port in nfs_ports:
+                    if each_port not in exist_ports:
+                        process.run("echo '%s' >> %s" %
+                                    (each_port, nfs_sysconfig), shell=True)
+                rpcd.restart()
+                nfsd.restart()
+                rule_temp = "INPUT -m state --state NEW -p %s -m multiport "
+                rule_temp += "--dport 111,892,2049,%s -s %s -j ACCEPT"
+                rule = rule_temp % ("tcp", tcp_port, subnet_mask)
+                rule_list.append(rule)
+                rule = rule_temp % ("udp", udp_port, subnet_mask)
+                rule_list.append(rule)
+                # Haven't imported libvirt as it creates cyclic import issue
+                # as nfs is already imported from libvirt.py
+                libvirt.setup_or_cleanup_iptables_rules(rule_list)
```

Comment:
So how can it be used if the module not import?
As config iptables' is not only for libvirt test(usually put in utils_test/libvirt.py), so maybe you can mv it to another module, or create a new module for iptables